noobaa/noobaa-core
# Complete GCP namespace support

### Explain the changes
1. This PR implements several functions (namely multipart uploads and tag support) required for proper namespace support over Google Cloud Storage.
I opted to use AWS SDK v3 for posterity, and went with barebones commands to speed things up a bit,
2. Since GCP only supports multipart uploads via the S3-compatible XML API, it is necessary to generate an HMAC key to use an S3 client with GCP. The PR also takes care of generating the HMAC key, as well as storing it in our DB.

### Issues: Fixed #xxx / Gap #xxx
1.  Up until now, our GCP interface supported resumable uploads but not multipart uploads - this PR addresses that gap

### Testing Instructions:
1. Use multipart upload to upload an object with tags to a GCP namespace bucket
3. Read the uploaded object and compare checksums to verify both write and read were successful
4. Read the object's tags and compare them to the ones from before the upload
5. Add additional tags to the object
6. Make sure the additional tags are present as well

I tested the PR locally with multipart uploads, as well as addition and removal of object metadata
